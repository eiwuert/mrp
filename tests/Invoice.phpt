<?php

use Tester\Assert;

require __DIR__ . '/bootstrap.php';

$mrp = new \Mrp\Invoice();
$mrp->addRow([
	'idfak' => '1',
	'druh' => 'F',
	'cislo' => '2017100001',
	'varsymb' => '2017100001',
	'ico' => '45580090',
	'typdph' => 41,
	'rezim_dph' => 0,
	'stat_dph' => 'SK',
	'vatnumber' => 'SK12345678',
	'kh_leasing' => 'F',
	'zakl0' => '1500', // price without VAT
	'zakl1' => 'AAAAAAAAAAAAAAAAAAA',
	'zakl2' => 'AAAAAAAAAAAAAAAAAAA',
	'mimodph' => 'AAAAAAAAAAAAAAAAAAA',
	'dph1' => 'AAAAAAAAAAAAAAAAAAA',
	'dph2' => 'AAAAAAAAAAAAAAAAAAA',
	'zaplaccislo' => 'AAAAAAAAAAAAAAAAAAA',
	'cislododli' => '2017100001',
	'storno_fa' => 'AAAAAAAAAAAAAAAAAAA',
	'storno_pro' => 'AAAAAAAAAAAAAAAAAAA',
	'dobropis_pro' => 'AAAAAAAAAAAAAAAAAAA',
	'vrubopis_pro' => 'AAAAAAAAAAAAAAAAAAA',
	'cis_predf' => 'AAAAAAAAAAAAAAAAAAA',
	'datvystave' => '1990-05-26',
	'datzdanpln' => '1990-05-26',
	'datsplatno' => '1990-05-26',
	'datdodani' => '1990-05-26',
	'konstsymb' => '0008',
	'specisymb' => 'AAAAAAAAAAAAAAAAAAA',
	'stredisko' => 'AAAAAAAAAAAAAAAAAAA',
	'formauhrad' => 'prevodom',
	'sposobdopr' => 'kurier',
	'cisloobjed' => '2017200059',
	'datobjed' => '1990-05-26',
	'prikazuhr' => 'AAAAAAAAAAAAAAAAAAA',
	'mena' => 'EUR',
	'kurz_zahr' => '1',
	'kurz_sk' => '1',
	'kurz_eur' => '1',
	'kurz_eur_p' => '1',
	'platkar' => 'AAAAAAAAAAAAAAAAAAA',
	'cisplatkar' => 'AAAAAAAAAAAAAAAAAAA',
	'typ_dokl' => 'AAAAAAAAAAAAAAAAAAA',
	'origcislo' => 'AAAAAAAAAAAAAAAAAAA',
	'cislo_zak' => 'AAAAAAAAAAAAAAAAAAA',
	'celk_zahr' => 'AAAAAAAAAAAAAAAAAAA',
	'icoprij' => 'AAAAAAAAAAAAAAAAAAA',
	'cenysdph' => 'F',
	'zlava' => 'AAAAAAAAAAAAAAAAAAA',
	'skonto' => 'AAAAAAAAAAAAAAAAAAA',
	'skontoproc' => 'AAAAAAAAAAAAAAAAAAA',
	'skontodny' => 'AAAAAAAAAAAAAAAAAAA',
	'kodplneni' => 'AAAAAAAAAAAAAAAAAAA',
	'hmotnost' => '10',
	'typ_pol' => 'AAAAAAAAAAAAAAAAAAA',
	'pdsyntet' => 'AAAAAAAAAAAAAAAAAAA',
	'pdanalyt' => 'AAAAAAAAAAAAAAAAAAA',
	'poznamka' => 'Put your note here',
	'spotrdand' => 'AAAAAAAAAAAAAAAAAAA',
	'ekodand' => 'AAAAAAAAAAAAAAAAAAA',
	'ekokom' => 'AAAAAAAAAAAAAAAAAAA',
	'usrfld1' => 'AAAAAAAAAAAAAAAAAAA',
	'usrfld2' => 'AAAAAAAAAAAAAAAAAAA',
	'usrfld3' => 'AAAAAAAAAAAAAAAAAAA',
	'usrfld4' => 'AAAAAAAAAAAAAAAAAAA',
	'usrfld5' => 'AAAAAAAAAAAAAAAAAAA',
]);

Assert::exception(function () use ($mrp) {
	$mrp->addRow(['idfak' => '1', 'cislo' => '2']);
}, 'Mrp\InvoiceException', "Missing field 'ico'");

Assert::exception(function () use ($mrp) {
	$mrp->addRow(['idfak' => '1', 'cislo' => '2', 'ico' => '3', 'unknown' => 'unknown']);
}, 'Mrp\InvoiceException', "Unknown field 'unknown'");

Assert::matchFile(__DIR__ . '/Invoice.getXml().expected', $mrp->getXml());
