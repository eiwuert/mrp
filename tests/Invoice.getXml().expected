<?xml version="1.0" encoding="windows-1250"?>
<document>
  <datasets>
    <dataset0>
      <rows>
        <row>
          <fields>
            <idfak>1</idfak>
            <druh>F</druh>
            <cislo>2017100001</cislo>
            <varsymb>2017100001</varsymb>
            <ico>45580090</ico>
            <typdph>41</typdph>
            <rezim_dph>0</rezim_dph>
            <stat_dph>SK</stat_dph>
            <vatnumber>SK12345678</vatnumber>
            <kh_leasing>F</kh_leasing>
            <zakl0>1500</zakl0>
            <zakl1>0</zakl1>
            <zakl2>0</zakl2>
            <mimodph>0</mimodph>
            <dph1>0</dph1>
            <dph2>0</dph2>
            <zaplaccislo>AAAAAAAAAA</zaplaccislo>
            <cislododli>2017100001</cislododli>
            <storno_fa>AAAAAAAAAA</storno_fa>
            <storno_pro>AAAAAAAAAA</storno_pro>
            <dobropis_pro>AAAAAAAAAA</dobropis_pro>
            <vrubopis_pro>AAAAAAAAAA</vrubopis_pro>
            <cis_predf>AAAAAAAAAA</cis_predf>
            <datvystave>1990-05-26</datvystave>
            <datzdanpln>1990-05-26</datzdanpln>
            <datsplatno>1990-05-26</datsplatno>
            <datdodani>1990-05-26</datdodani>
            <konstsymb>0008</konstsymb>
            <specisymb>AAAAAAAAAA</specisymb>
            <stredisko>AAAAAA</stredisko>
            <formauhrad>prevodom</formauhrad>
            <sposobdopr>kurier</sposobdopr>
            <cisloobjed>2017200059</cisloobjed>
            <datobjed>1990-05-26</datobjed>
            <prikazuhr>0</prikazuhr>
            <mena>EUR</mena>
            <kurz_zahr>1</kurz_zahr>
            <kurz_sk>1</kurz_sk>
            <kurz_eur>1</kurz_eur>
            <kurz_eur_p>1</kurz_eur_p>
            <platkar>0</platkar>
            <cisplatkar>AAAAAAAAAAAAAAAAAAA</cisplatkar>
            <typ_dokl>AA</typ_dokl>
            <origcislo>AAAAAAAAAAAAAAAAAAA</origcislo>
            <cislo_zak>AAAAAAAAAAAAAAA</cislo_zak>
            <celk_zahr>0</celk_zahr>
            <icoprij>AAAAAAAAAAAA</icoprij>
            <cenysdph>F</cenysdph>
            <zlava>0</zlava>
            <skonto>0</skonto>
            <skontoproc>0</skontoproc>
            <skontodny>0</skontodny>
            <kodplneni>A</kodplneni>
            <hmotnost>10</hmotnost>
            <typ_pol>AA</typ_pol>
            <pdsyntet>0</pdsyntet>
            <pdanalyt>0</pdanalyt>
            <poznamka>Put your note here</poznamka>
            <spotrdand>0</spotrdand>
            <ekodand>0</ekodand>
            <ekokom>0</ekokom>
            <usrfld1>AAAAAAAAAAAAAAAAAAA</usrfld1>
            <usrfld2>AAAAAAAAAAAAAAAAAAA</usrfld2>
            <usrfld3>AAAAAAAAAAAAAAAAAAA</usrfld3>
            <usrfld4>AAAAAAAAAAAAAAAAAAA</usrfld4>
            <usrfld5>AAAAAAAAAAAAAAAAAAA</usrfld5>
          </fields>
        </row>
      </rows>
    </dataset0>
  </datasets>
</document>
